src/main/java/uk/gov/justice/digital/delius/service/AttendanceService.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.justice.digital.delius.service;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.digital.delius.data.api.Attendance;
+import uk.gov.justice.digital.delius.data.api.Attendance.ContactTypeDetail;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Contact;
+import uk.gov.justice.digital.delius.jpa.standard.repository.ContactRepository;
+
+@Slf4j
+@Service
+public class AttendanceService {
+
+    private final ContactRepository contactRepository;
+
+    @Autowired
+    public AttendanceService(final ContactRepository contactRepository) {
+        this.contactRepository = contactRepository;
+    }
+
+    public List<Contact> getContactsForEvent(final Long offenderId, final Long eventId, final LocalDate localDate) {
+        return contactRepository.findByOffenderAndEventIdEnforcement(offenderId, eventId, localDate);
+    }
+
+    static boolean forEntityBoolean(final String booleanStr) {
+        if (booleanStr == null)
+            return false;
+        return booleanStr.trim().equals("1");
+    }
+
+    public static List<Attendance> attendancesFor(final List<Contact> contacts) {
+
+        if (contacts == null) {
```

Comment:
BTW I'm pretty certain you would get a empty list when none found rather than null list

Reply:
Leaving